bolt-sidecar/src/state/execution.rs

```diff
@@ -326,12 +337,30 @@ impl<C: StateFetcher> ExecutionState<C> {
                 }
             };
 
+            let sender_nonce_diff = bundle_nonce_diff_map.entry(sender).or_insert(0);
+            let sender_balance_diff = bundle_balance_diff_map.entry(sender).or_insert(U256::ZERO);
+
+            // Apply the diffs to this account according to the info fetched from the templates
+            // and the current bundle diffs for this sender.
             let account_state_with_diffs = AccountState {
-                transaction_count: account_state.transaction_count.saturating_add(nonce_diff),
-                balance: account_state.balance.saturating_sub(balance_diff),
+                transaction_count: account_state
+                    .transaction_count
+                    .saturating_add(nonce_diff)
+                    .saturating_add(*sender_nonce_diff),
+
+                balance: account_state
+                    .balance
+                    .saturating_sub(balance_diff)
+                    .saturating_sub(*sender_balance_diff),
+
+                // TODO(nico): what if this changes in the middle of the request?
                 has_code: account_state.has_code,
```

Comment:
I don't think this can happen even with EIP-7702. What are your doubts about it?

Reply:
Yeah, after thinking more about it, I am convinced it can't happen

Reply:
removed